streams/src/test/java/org/apache/kafka/streams/kstream/JoinWindowsTest.java

```diff
@@ -102,4 +103,94 @@ public void gracePeriodShouldEnforceBoundaries() {
         }
     }
 
+    @Test
+    public void equalsAndHashcodeShouldBeValidForPositiveCases() {
+        assertEquals(JoinWindows.of(3), JoinWindows.of(3));
```

Comment:
personally, I prefer using a library to test this: 

http://static.javadoc.io/com.google.guava/guava-testlib/21.0/com/google/common/testing/EqualsTester.html

Reply:
AK does not yet have a guava dependency I think?

Reply:
Nice! Guozhang is right, we don't have that dependency, but I could write a test util to check the same conditions. I'll plan to do this.

Reply:
@vvcephei this isn't in guava, it's in guava-testlib, which is a test only dependency. So, I don't see why this couldn't be added as a dependency, rather than writing our own.